pkg/utils/probe_helper.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+// DeploymentProber is to check the addon status based on status
+// of the agent deployment status
+type DeploymentProber struct {
+	deployments []types.NamespacedName
+}
+
+func NewDeploymentProber(deployments ...types.NamespacedName) *agent.HealthProber {
+	return &agent.HealthProber{
+		Type: agent.HealthProberTypeWork,
+		WorkProber: &DeploymentProber{
+			deployments: deployments,
+		},
+	}
+}
+
+func (d *DeploymentProber) ProbeFields() []agent.ProbeField {
+	probeFields := []agent.ProbeField{}
+	for _, deploy := range d.deployments {
+		probeFields = append(probeFields, agent.ProbeField{
+			ResourceIdentifier: workapiv1.ResourceIdentifier{
+				Group:     "apps",
+				Resource:  "deployments",
+				Name:      deploy.Name,
+				Namespace: deploy.Namespace,
+			},
+			ProbeRules: []workapiv1.FeedbackRule{
+				{
+					Type: workapiv1.WellKnownStatusType,
+				},
+			},
+		})
+	}
+	return probeFields
+}
+
+func (d *DeploymentProber) HealthCheck(identifier workapiv1.ResourceIdentifier, result workapiv1.StatusFeedbackResult) error {
```

Comment:
the per-resource conditions provides a status named "StatusFeedbackSynced", how do you think about checking the overall condition before digging into the fields?

Reply:
here is a part to check the condition before checking using check func https://github.com/open-cluster-management-io/addon-framework/pull/77/files#diff-6cbce2e82f874dcc6e37cfb90ab66b42db275761808cca2abfd5c8b294d67e11R172. I will add a comment on that